// @httpx/exception/dist/index.cjs
'use strict';

var chunkU5YYVW4G_cjs = require('./chunk-U5YYVW4G.cjs');
var chunkD5VRF2Z5_cjs = require('./chunk-D5VRF2Z5.cjs');
var chunkMIP7DPOM_cjs = require('./chunk-MIP7DPOM.cjs');

var m=(t,o=!0)=>t instanceof chunkMIP7DPOM_cjs.c&&(!o||t.statusCode>399&&t.statusCode<500);var u=(t,o=!0)=>t instanceof chunkMIP7DPOM_cjs.d&&(!o||t.statusCode>499&&t.statusCode<600);var E=t=>typeof t=="number"&&t>99&&t<600;

Object.defineProperty(exports, "isErrorWithErrorStatusCode", {
	enumerable: true,
	get: function () { return chunkU5YYVW4G_cjs.a; }
});
Object.defineProperty(exports, "isObjectWithErrorStatusCode", {
	enumerable: true,
	get: function () { return chunkU5YYVW4G_cjs.b; }
});
Object.defineProperty(exports, "isHttpException", {
	enumerable: true,
	get: function () { return chunkD5VRF2Z5_cjs.a; }
});
Object.defineProperty(exports, "HttpBadGateway", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.H; }
});
Object.defineProperty(exports, "HttpBadRequest", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.e; }
});
Object.defineProperty(exports, "HttpClientException", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.c; }
});
Object.defineProperty(exports, "HttpConflict", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.f; }
});
Object.defineProperty(exports, "HttpException", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.b; }
});
Object.defineProperty(exports, "HttpExpectationFailed", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.g; }
});
Object.defineProperty(exports, "HttpFailedDependency", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.h; }
});
Object.defineProperty(exports, "HttpForbidden", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.i; }
});
Object.defineProperty(exports, "HttpGatewayTimeout", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.I; }
});
Object.defineProperty(exports, "HttpGone", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.j; }
});
Object.defineProperty(exports, "HttpImATeapot", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.k; }
});
Object.defineProperty(exports, "HttpInsufficientStorage", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.J; }
});
Object.defineProperty(exports, "HttpInternalServerError", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.K; }
});
Object.defineProperty(exports, "HttpLengthRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.l; }
});
Object.defineProperty(exports, "HttpLocked", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.m; }
});
Object.defineProperty(exports, "HttpLoopDetected", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.L; }
});
Object.defineProperty(exports, "HttpMethodNotAllowed", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.n; }
});
Object.defineProperty(exports, "HttpMisdirectedRequest", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.o; }
});
Object.defineProperty(exports, "HttpNetworkAuthenticationRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.M; }
});
Object.defineProperty(exports, "HttpNotAcceptable", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.p; }
});
Object.defineProperty(exports, "HttpNotExtended", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.N; }
});
Object.defineProperty(exports, "HttpNotFound", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.q; }
});
Object.defineProperty(exports, "HttpNotImplemented", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.O; }
});
Object.defineProperty(exports, "HttpPayloadTooLarge", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.r; }
});
Object.defineProperty(exports, "HttpPaymentRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.s; }
});
Object.defineProperty(exports, "HttpPreconditionFailed", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.t; }
});
Object.defineProperty(exports, "HttpPreconditionRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.u; }
});
Object.defineProperty(exports, "HttpProxyAuthenticationRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.v; }
});
Object.defineProperty(exports, "HttpRangeNotSatisfiable", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.w; }
});
Object.defineProperty(exports, "HttpRequestHeaderFieldsTooLarge", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.x; }
});
Object.defineProperty(exports, "HttpRequestTimeout", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.y; }
});
Object.defineProperty(exports, "HttpServerException", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.d; }
});
Object.defineProperty(exports, "HttpServiceUnavailable", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.P; }
});
Object.defineProperty(exports, "HttpTooEarly", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.z; }
});
Object.defineProperty(exports, "HttpTooManyRequests", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.A; }
});
Object.defineProperty(exports, "HttpUnauthorized", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.B; }
});
Object.defineProperty(exports, "HttpUnavailableForLegalReasons", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.C; }
});
Object.defineProperty(exports, "HttpUnprocessableEntity", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.D; }
});
Object.defineProperty(exports, "HttpUnsupportedMediaType", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.E; }
});
Object.defineProperty(exports, "HttpUpgradeRequired", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.F; }
});
Object.defineProperty(exports, "HttpUriTooLong", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.G; }
});
Object.defineProperty(exports, "HttpVariantAlsoNegotiates", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.Q; }
});
Object.defineProperty(exports, "HttpVersionNotSupported", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.R; }
});
Object.defineProperty(exports, "createHttpException", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.T; }
});
Object.defineProperty(exports, "isHttpErrorStatusCode", {
	enumerable: true,
	get: function () { return chunkMIP7DPOM_cjs.S; }
});
exports.isHttpClientException = m;
exports.isHttpServerException = u;
exports.isHttpStatusCode = E;
